app/services/gitlab_repo_service.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'rest-client'
+require 'json'
+require 'uri'
+
+class GitlabRepoService
+  def initialize(user)
+    @base_url = 'https://gitlab.com/api/'
+    @api_version = 'v4/'
+    @user_projects_endpoint = "users/#{user.gid}/projects"
+    @per_page = 9
```

Comment:
Use initializer for setting state. Anything that is not related to the state of the class should be a constant

Reply:
aha, been a while since I wrote classes! 😄
changed!